src/frontend/Deprecation_analysis.ml

```diff
@@ -137,6 +137,40 @@ let rec collect_deprecated_expr (acc : (Location_span.t * string) list)
                 ) ]
           | _ -> [] ) in
       acc @ w @ List.concat_map l ~f:(fun e -> collect_deprecated_expr [] e)
+  | PrefixOp (PNot, ({emeta= {type_= UReal; loc; _}; _} as e)) ->
+      let acc =
+        acc
+        @ [ ( loc
+            , "Using a real as a boolean value is deprecated and will be \
+               disallowed in Stan 2.34. Use an explicit != 0 comparison \
+               instead." ) ] in
```

Comment:
We should also note that the canonicalizer can rewrite this for you, like we do for the other warnings

Reply:
We don't say that for binary op warnings (int-division and matrix-power) which this was modeled after but I'm fine with adding such a comment here.

Reply:
Those don’t say so because it (I believe) the canonicalizer doesn’t do anything for them, correct?

Reply:
Oh no, you're right. Canonicalizer definitely should change int division to `%/%`, no?